frontends/bmd/src/utils/ScreenReader/kiosk_text_to_speech.ts

```diff
@@ -0,0 +1,142 @@
+import { SpeakOptions, TextToSpeech } from '../../config/types';
+
+interface SpeechItem {
+  utterance: string;
+  resolve: (value: void | PromiseLike<void>) => void;
+}
+
+export class KioskTextToSpeech implements TextToSpeech {
+  private readonly speechQueue: SpeechItem[] = [];
+  private isSpeaking = false;
+
+  // Simultaneous calls to kiosk.cancelSpeak will cause duplicated spd-say
+  // calls to fail. We avoid conflicts by tracking the corresponding promise.
+  private promiseToStop?: Promise<void> = undefined;
+
+  private muted = false;
+  private readonly volume = 50;
+
+  constructor(initiallyMuted: boolean) {
+    this.muted = initiallyMuted;
+  }
+  /**
+   * Queues utterance to be sent to the speech dispatcher
+   */
+  async speak(
+    utterance: string,
+    { now = false }: SpeakOptions = {}
+  ): Promise<void> {
+    if (this.isMuted()) {
+      return;
+    }
+
+    if (now) {
+      await this.stop();
+    }
+
+    await this.enqueueSpeech(utterance);
```

Comment:
I see why you chose to do this, which is to keep supporting the idea that `speak()` supports the idea of queuing up things to speak... but I'm torn because this adds a good bit of complexity and state for a feature that I don't think we ever use, nor do I anticipate we will want to use it – I can't think of a use case in our context.

If we want to support this feature, I think it's best supported in `kiosk-browser`, where we would pass the `now` argument, and implement it there. And for now, I would suggest we simply error out on `now=false` and say "not supported yet." We leave ourselves the API opportunity in case we do find a use case for it, but we don't take on the complexity and associated bugs (there will be bugs!)